src/utils/addNetworkPrefix.ts

```diff
@@ -0,0 +1,21 @@
+import { extractShortChainName } from 'src/routes/routes'
+import getNetworkPrefix from './getNetworkPrefix'
+
+function addNetworkPrefix(address = '', showNetworkPrefix = true, netWorkPrefix = extractShortChainName()): string {
+  // if showNetworkPrefix is disabled in appearance settings we return the address as it is
+  if (!showNetworkPrefix) {
+    return address
+  }
+
+  const prefix = getNetworkPrefix(address)
```

Comment:
You can use `getCurrentShortChainName()` for this.

Reply:
`getCurrentShortChainName` always returns the shortName of the current selected network, but the address that the user types in the input could be could be incorrect.

Reply:
Also an oversight. Maybe it's good to use `isValidShortChainName` to parse it before returning it.